File: frontend/src/types/Deployment.ts
export interface Deployment {
  id: number;
  environment: 'production' | 'staging' | 'dev';
  status: 'success' | 'failure' | 'running' | 'pending';
  commit: string;
  triggeredBy: string;
  duration: number;
  deployedAt: string;
}

File: internal/api/rest/response/deployment.go
package response

type Deployment struct {
	ID          uint   `json:"id"`
	AppID       uint   `json:"appId"`
	Environment string `json:"environment"`
	Status      string `json:"status"`
	Commit      string `json:"commit"`
	TriggeredBy string `json:"triggeredBy"`
	Duration    int64  `json:"duration"`
	DeployedAt  string `json:"deployedAt"`
}
